airflow/hooks/postgres_hook.py

```diff
@@ -67,6 +68,13 @@ def get_conn(self):
             password=conn.password,
             dbname=self.schema or conn.schema,
             port=conn.port)
+        if conn.extra_dejson.get('cursor', False):
+            if (conn.extra_dejson['cursor']).lower() == 'dictcursor':
+                conn_args['cursor_factory'] = psycopg2.extras.DictCursor
+            elif (conn.extra_dejson['cursor']).lower() == 'realdictcursor':
+                conn_args['cursor_factory'] = psycopg2.extras.RealDictCursor
+            elif (conn.extra_dejson['cursor']).lower() == 'namedtuplecursor':
+                conn_args['cursor_factory'] = psycopg2.extras.NamedTupleCursor
```

Comment:
This looks like a [factory pattern](https://en.wikipedia.org/wiki/Factory_method_pattern) can be of good use here :)

```python
def _get_cursor(self):
    _cursor = conn.extra_dejson['cursor'].lower()
    if _cursor == 'dictcursor':
        return psycopg2.extras.DictCursor
    if _cursor == 'realdictcursor':
        return psycopg2.extras.RealDictCursor
    if _cursor == 'namedtuplecursor':
        return psycopg2.extras.NamedTupleCursor
    raise ValueError('Invalid cursor passed {}'.format(_cursor ))
```

I also added complete validation (see the last line).

WDYT?
So after you have this factory you can just call it like that.
```python
if conn.extra_dejson.get('cursor', False):
    conn_args['cursor_factory'] = self._get_cursor()
```

Reply:
That is a great pointing. I'd love to apply your suggestions.

And, I think we can also use this pattern for the mysql hook as well.

Reply:
Are there some conventions for the order of public/private methods in a class?

Reply:
Hm, wait. `conn` is a local variable for `get_conn` method. So we should get the connection in the `_get_cursor` again by

```python
conn_id = getattr(self, self.conn_name_attr)
conn = self.get_connection(conn_id)
```

So `get_connection` will be called twice when cursor class is specified. Is it ok?

Reply:
```
def _get_cursor(self, raw_cursor):
    _cursor = raw_cursor.lower()
    if _cursor == 'dictcursor':
        return psycopg2.extras.DictCursor
    if _cursor == 'realdictcursor':
        return psycopg2.extras.RealDictCursor
    if _cursor == 'namedtuplecursor':
        return psycopg2.extras.NamedTupleCursor
    raise ValueError('Invalid cursor passed {}'.format(_cursor))
```

```
raw_cursor = conn.extra_dejson.get('cursor', False)
if raw_cursor:
    conn_args['cursor_factory'] = self._get_cursor(raw_cursor)
```

Reply:
I updated the code to use the cursor factory and added an additional test case.